src/components/AngerNoodle/helperFunctions/gameLogic.js

```diff
@@ -0,0 +1,110 @@
+import { makeRandomPositionIndex, copyBoard, copySnake, angryMessages } from "./data";
+
+const computerNextFrame = (shouldDeleteTail, updateShouldDeleteTail, snake, updateSnake, gridSize, board, updateBoard, direction, wallsAreLava, treatCoords, updateTreatCoords, currentScore, updateScore, highScore, updateHighScore, gameStateRef, snakeRef, updateAngryMessage) => {
+  let score = currentScore;
+  const currentHighScore = highScore;
+  let newTreatCoords = null;
+  const gameState = gameStateRef;
+  const currentSnakeRef = snakeRef;
+  const boardCopy = copyBoard(board);
+  const snakeCopy = copySnake(snake);
+  const head = snakeCopy[snake.length - 1];
+  const boundary = gridSize - 1;
+
+  const removeTail = () => {
+    snakeCopy.splice(0, 1);
+  };
+
+  const changeDirection = () => {
+    const [headX, headY] = head;
+    // console.log(`head x: ${headX}, head y: ${headY}`);
+    let newHeadPosition;
+    switch (direction) {
+      case 'up':
+        newHeadPosition = headX !== 0 ? [headX - 1, headY] : [boundary, headY];
+        break;
+      case 'down':
+        newHeadPosition = headX !== boundary ? [headX + 1, headY] : [0, headY];
+        break;
+      case 'left':
+        newHeadPosition = headY !== 0 ? [headX, headY - 1] : [headX, boundary];
+        break;
+      case 'right':
+        newHeadPosition = headY !== boundary ? [headX, headY + 1] : [headX, 0];
+        break;
+      default:
+        break;
+    }
+    return newHeadPosition;
+  };
+
+  const checkNextMove = (nextHeadPosition) => {
+    const [nextHeadPositionY, nextHeadPositionX] = nextHeadPosition;
+    const nextHeadCell = boardCopy[nextHeadPositionY][nextHeadPositionX];
+    const aboutToHitTopWall =
+      direction === "up" && nextHeadPositionY === boundary;
+    const aboutToBottomWall = direction === "down" && nextHeadPositionY === 0;
+    const aboutToHitLeftWall =
+      direction === "left" && nextHeadPositionX === boundary;
+    const aboutToHitRightWall =
+      direction === "right" && nextHeadPositionX === 0;
+    const bonk = wallsAreLava &&
+      (aboutToHitTopWall ||
+        aboutToBottomWall ||
+        aboutToHitLeftWall ||
+        aboutToHitRightWall);
+
+    if (nextHeadCell.classString.includes("segment") || bonk) {
+      gameState.current = 'gameOver';
+    }
+    if (nextHeadCell.classString.includes("treat")) {
```

Comment:
Instead of checking the class string I'd recommend doing some quick math using the variables you have - some options would be:
- using your treatLocation chunk of state and comparing it to the head location
- storing your "treat" on the "board" object the same way you are with classStrings

Reply:
Switched to storing `isSegment` and `isTreat` on the board object cell objects.